client/src/main/scala/io/delta/sharing/client/util/RetryUtils.scala

```diff
@@ -65,10 +63,10 @@ private[sharing] object RetryUtils extends Logging {
 
   def shouldRetry(t: Throwable): Boolean = {
     t match {
-      case e: UnexpectedHttpStatus =>
-        if (e.statusCode == 429) { // Too Many Requests
+      case DeltaSharingExceptionWithErrorCode(Some(statusCode)) =>
```

Comment:
and could it be a DeltaSharingExceptionWithErrorCode with `None` as statusCodeOpt? What if that, we don't retry?

Reply:
is it tested?

Reply:
cool, thanks!